### PR TITLE
System.IO.Path.IsPathRooted tests added

### PR DESCRIPTION
**Pull request by ViktorHofer**

### CODE REVIEW

File: src/System.IO.FileSystem/tests/Path/GetPathRoot.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Path_GetPathRoot : FileSystemTest
+    {
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/Path/GetPathRoot.cs]: we already have some GetPathRoot tests in PathTests.cs. Merge in here?

File: src/System.IO.FileSystem/tests/Path/GetPathRoot.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        [Fact]
+        public void NullAsPath_ReturnsNull()
+        {
+            Assert.Null(Path.GetPathRoot(null));
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/Path/GetPathRoot.cs]: try also > 260 characters of whitespace

Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/Path/GetPathRoot.cs]: embedded null? https://referencesource.microsoft.com/#mscorlib/system/io/path.cs,493

Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/Path/GetPathRoot.cs]: invalid chars also seems a missing case

Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/Path/GetPathRoot.cs]: same for IsPathRooted

File: src/System.IO.FileSystem/tests/Path/GetPathRoot.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/Path/GetPathRoot.cs]: same for merging IsPathRooted into your other file

**Comment by danmosemsft** [comment]
obiously please run all these on desktop also ...

are no quirks relevant here - need checking for? not sure how one tests both ways, @safern would know.

File: src/System.IO.FileSystem/tests/Path/GetPathRoot.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Path_GetPathRoot : FileSystemTest
+    {
+        [Fact]
+        public void NullAsPath_ReturnsNull()
+        {
+            Assert.Null(Path.GetPathRoot(null));
```
Comment (safern) [inline src/System.IO.FileSystem/tests/Path/GetPathRoot.cs]: We already have a test for this: https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/tests/System/IO/PathTests.cs#L177

File: src/System.IO.FileSystem/tests/Path/GetPathRoot.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
+        public void IllegalPath_ThrowsArgumentException(string path)
+        {
```
Comment (safern) [inline src/System.IO.FileSystem/tests/Path/GetPathRoot.cs]: I'm already adding tests on: https://github.com/dotnet/corefx/pull/19343

We could keep this one, or the other, whichever you prefer.

File: src/System.IO.FileSystem/tests/Path/IsPathRooted.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Path_IsPathRooted : FileSystemTest
+    {
```
Comment (safern) [inline src/System.IO.FileSystem/tests/Path/IsPathRooted.cs]: This tests doesn't exist, could you please add them instead on PathTests.cs as @danmosemsft mentioned before? 

https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

**Comment by safern** [comment]
> are no quirks relevant here - need checking for? not sure how one tests both ways, @safern would know.

We already have some helper methods that check if a quirk is enable so in that way we could test both ways by conditioning the test. Helper methods are in: https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/IO/PathFeatures.cs

We have methods that check for Legacy Path and Long Path quirks.

**Comment by safern** [comment]
cc: @JeremyKuhne @ianhays

**Comment by danmosemsft** [comment]
@safern am I right that: in the lab, our desktop runs run against 4.6.1 TFM, on the 4.7.1 framework? 

What about on a dev machine, do I get 4.6.1. TFM but on whatever framework is installed? 

If a bug was fixed in 4.6.2 (like some Crypto ones we found) without a quirk around it, both runs and devs should see it?

**Comment by ViktorHofer** [comment]
What a mess... Sry for that, fixed now.

@danmosemsft I just wanted to add tests which cover the changes I made in coreclr. You are absolutely right, there are code paths which aren't covered yet by any tests but that's not something I wanna do in this PR.

**Comment by safern** [comment]
> @safern am I right that: in the lab, our desktop runs run against 4.6.1 TFM, on the 4.7.1 framework?

Nope all the tests run against TFM 4.5.2 which is Xunit's TFM and Xunit is the executing AppDomain. 

I'm not sure what framework version the lab machines have installed, but we should update them all at least to 4.7

> What about on a dev machine, do I get 4.6.1. TFM but on whatever framework is installed?

TFM version will always depend on Xunit's TFM (which is unfortunate). I have issue: https://github.com/dotnet/corefx/issues/15136 tracking this, which I will investigate after having a clean run enabled on CI.

> If a bug was fixed in 4.6.2 (like some Crypto ones we found) without a quirk around it, both runs and devs should see it?

Fixes without a quirk will depend on the latest framework installed on the running machine. So we could condition them with PlatformDetection helper APIs that parse the running version. We have:

`Is462OrNewer()` and `Is47OrNewer()`. We can add whichever we want as we already have a method parsing current running version, we just need to compare that against the version we are interested in.

**Comment by ViktorHofer** [comment]
@safern I removed my GetPathRoot tests in favor of your PR. So what is left in this PR is just adding a null, empty string and whitespace string for IsPathRooted.

So this isn't connected to the coreclr fix anymore...

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs [hunk 6]
```diff
@@ -237,10 +237,18 @@ public static void GetPathRoot_Unix()
...
+        public static void IsPathRooted()
+        {
+            Assert.False(Path.IsPathRooted(null));
+        }
+
         // Testing invalid drive letters !(a-zA-Z)
         [PlatformSpecific(TestPlatforms.Windows)]
         [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
```
Comment (safern) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: This two cases return false as well on Desktop and this tests is skipped on Net Framework. Could we instead move this test cases to `IsPathRooted()` test that you are adding above? Make it a theory and pass as inline data:
```cs
[InlineData("")]
[InlineData(" ")]
[InlineData(null)]
```

**Comment by safern** [comment]
> @safern I removed my GetPathRoot tests in favor of your PR. So what is left in this PR is just adding a null, empty string and whitespace string test for IsPathRooted.

Thanks!

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs (hunk 6 above)
Comment (ViktorHofer) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: Absolutely. As this PR now contains little to nothing I will hold it for now and add some more test cases later today.

**Pull request merged**